plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1016,14 +1028,22 @@ protected JdbcOutputTableHandle beginInsertTable(
         String remoteTemporaryTableName = identifierMapping.toRemoteTableName(remoteIdentifiers, identity, remoteSchema, generateTemporaryTableName(session));
         copyTableSchema(session, connection, catalog, remoteSchema, remoteTable, remoteTemporaryTableName, columnNames.build());
 
+        RemoteTableName remoteTableTemporaryTable = new RemoteTableName(
+                Optional.ofNullable(catalog),
+                Optional.ofNullable(remoteSchema),
+                remoteTemporaryTableName);
+
+        for (JdbcColumnHandle column : additionalColumns) {
+            addColumn(session, connection, remoteTableTemporaryTable, column.getColumnMetadata());
```

Comment:
why `ALTER TABLE ADD`. Extra roundtrips to database seem not to be best approach. Why not building final schema and just do single `CREATE TABLE`?